### PR TITLE
Cleanup code a little.

### PR DESCRIPTION
**Pull request by MatthewDolan**

### CODE REVIEW

File: src/main/java/com/lightstep/tracer/shared/TextMapPropagator.java [hunk 1]
```diff
@@ -16,53 +18,40 @@
...
-        int requiredFieldCount = 0;
-
-        // traceId and spanId initial values are guaranteed to be overwritten because of the
-        // requiredFieldCount check
-        long traceId = 0;
-        long spanId = 0;
-        Map<String, String> decodedBaggage = null;
+        Long traceId = null;
+        Long spanId = null;
+        Map<String, String> baggage = new HashMap<>();
```
Comment (danielamiao) [inline src/main/java/com/lightstep/tracer/shared/TextMapPropagator.java]: Not a real comment, just want to note that reading this refactor makes me laugh, code was pretty convoluted for no reason, 👍  to improvements!

File: src/main/java/com/lightstep/tracer/shared/NoopSpan.java [hunk 2]
```diff
@@ -8,25 +8,21 @@
 
 // A span which is returned when the tracer is disabled.
 class NoopSpan implements Span {
-
     static final Span INSTANCE = new NoopSpan();
     private static final SpanContext CONTEXT = new NoopSpanContext();
 
-    private NoopSpan() {
-    }
+    private NoopSpan() {}
```
[the remark this answers is not in the document]
Comment (MatthewDolan) [reply src/main/java/com/lightstep/tracer/shared/NoopSpan.java]: No...that was all by me by hand. I am sure we could get IntelliJ to do this for us though.

File: src/main/java/com/lightstep/tracer/shared/SpanContext.java [hunk 3]
```diff
@@ -1,75 +1,78 @@
...
-        Map<String, String> baggageCopy;
-        if (ctxBuilder.getBaggageMap() != null) {
-            baggageCopy = new HashMap<>(ctxBuilder.getBaggageMap());
-        } else {
-            baggageCopy = new HashMap<>();
-        }
-        baggageCopy.put(key, value);
-        ctxBuilder.putBaggage(key, value);
-        return new SpanContext(ctxBuilder.getTraceId(), ctxBuilder.getSpanId(), baggageCopy);
+        // This is really a "set" not a "with" but keeping as is to preserve behavior.
```
[the remark this answers is not in the document]
Comment (MatthewDolan) [reply src/main/java/com/lightstep/tracer/shared/SpanContext.java]: I usually associate `withFoo(...)` with [persistent data structures](https://en.wikipedia.org/wiki/Persistent_data_structure) as opposed to setters which are sometimes persistent but often they just mutate the underlying state.

Here we are mutating the underlying state and also returning a new version of the object. Ideally we could stop mutating but I can't be sure that clients don't rely on that behavior.

File: src/main/java/com/lightstep/tracer/shared/TextMapPropagator.java (hunk 1 above)
Comment (MatthewDolan) [reply src/main/java/com/lightstep/tracer/shared/TextMapPropagator.java]: 😄

**Pull request merged**